xcm/xcm-emulator/src/lib.rs

```diff
@@ -364,52 +408,118 @@ macro_rules! __impl_test_ext_for_relay_chain {
 	// entry point: generate ext name
 	($name:ident, $genesis:expr, $on_init:expr, $api_version:tt) => {
 		$crate::paste::paste! {
-			$crate::__impl_test_ext_for_relay_chain!(@impl $name, $genesis, $on_init, [<ParachainHostV $api_version>], [<EXT_ $name:upper>]);
+			$crate::__impl_test_ext_for_relay_chain!(
+				@impl $name,
+				$genesis,
+				$on_init,
+				[<ParachainHostV $api_version>],
+				[<LOCAL_EXT_ $name:upper>],
+				[<GLOBAL_EXT_ $name:upper>]
+			);
 		}
 	};
 	// impl
-	(@impl $name:ident, $genesis:expr, $on_init:expr, $api_version:ident, $ext_name:ident) => {
+	(@impl $name:ident, $genesis:expr, $on_init:expr, $api_version:ident, $local_ext:ident, $global_ext:ident) => {
 		thread_local! {
-			pub static $ext_name: $crate::RefCell<$crate::sp_io::TestExternalities>
+			pub static $local_ext: $crate::RefCell<$crate::TestExternalities>
 				= $crate::RefCell::new(<$name>::build_new_ext($genesis));
 		}
 
+		$crate::lazy_static! {
```

Comment:
Can we use the newly stabilised `OnceLock` here?

Reply:
Can it be done in different PR? I wouldn't like to block this one for long, need to get my hands into something more urgent. If it is an easy fix please push the changes.